docs/implicit-provisioning.adoc

```diff
@@ -0,0 +1,34 @@
+# Implicit Provisioning
+
+This document describes the current support for implicit provisioning in Aktualizr and meta-updater.
+
+## Background
+
+In automatic provisioning, bootstrap credentials are downloaded and installed on devices to initiate the provisioning process. In implicit provisioning, devices are to be pre-loaded with the requisite provisioning credentials.
+
+## Configuration
+
+The following items are relevant for implicit provisioning:
+
+[options=header]
+|===================
+| Configuration option | Where it will come from
+| Server URL | Written into `sota.toml` during bitbake via `implicit_writer`
+| Root CA cert | Installed to `/usr/lib/sota/root.crt` during bitbake via `implicit_writer`
+| TLS key/cert | Pre-installed or manually installed, e.g. with `cert_provider`
+| Device ID | Read from Common Name field of TLS cert
+| Uptane public/private key | Automatically generated by Aktualizr
+| Uptane primary serial number | Automatically generated by Aktualizr
+| Primary ECU Hardware ID | Automatically generated by Aktualizr
+|===================
+
+An example `.toml` configuration file can be found at `config/sota_implicit_prov.toml`. This is what is used as in input to `implicit_writer` during bitbaking with meta-updater.
+
+## Steps
+
+1. In meta-updater, set `SOTA_CLIENT_PROV = "aktualizr-implicit-prov"`. Currently, you will also need to set `SOTA_PACKED_CREDENTIALS` to provisioning credentials zip file.
+1. Build a standard image using bitbake.
+1. Boot the image.
+1. Optionally, verify that Aktualizr is not provisioning and that the device is not visible in the Garage.
+1. Load credentials onto the device, for example with `cert_provider` from the Aktualizr repo: `cert_provider -c credentials.zip -t <device> -d /var/sota`
```

Comment:
It still looks like a description of hacky implicit provisioning: in the production case cert_provider is not used.

Reply:
What would you like to see to make it less hacky? After all, `cert_provider` is the only thing that works right now other than manually copying, which was the original "hacky" name referred to.

Reply:
Real implicit provisioning should be already enabled at least on CI, it might be time (next sprint probably) to create a manual on how to create devices for it with and without HSM. Until then I'm fine with keeping instructions for cert_provider in place, but IMO some note telling that this is not how implicit provisioning really works might be helpful.

Reply:
> Real implicit provisioning

Do you mean using an HSM? I didn't write a document for that case, although I certainly could. This document was just meant to describe how to make use of the `aktualizr-implicit-prov` recipe.

Either way, you are probably right that some better context of the intended use case is a good idea.

Reply:
Either HSM or storing pkey/ca/cert on the file system, but without using autoprovisioning. I mean the "normal" (at least it was our plan) process of creating root keypair, uploading public key to the backend and signing all the devices' individual certificates with the private key. It's what our testing people emulate as far as I understand.